src/EnergyPlus/ThermalComfort.cc

```diff
@@ -1985,35 +1807,94 @@ namespace ThermalComfort {
 		}
 
-		CalcAngleFactorMRT = SurfTempAngleFacSummed;
+		CalcAngleFactorMRT = root_4(SurfTempEmissAngleFacSummed/thisAngFacList.SumSurfaceEmissAngleFactor) - KelvinConv;
 
 		return CalcAngleFactorMRT;
 
 	}
 
+	Real64
+	CalcSurfaceWeightedMRT(
+		int const ZoneNum,
+		int const SurfNum
+	)
+	{
+		
+		// Purpose: Calculate a modified zone MRT that excludes the Surface( SurfNum ).
+		//          This is necessary for the surface weighted option to not in essence
+		//          double count SurfNum in the MRT calculation.  Other than that, the
+		//          method here is the same as CalculateZoneMRT.
+		
+		// Using/Aliasing
+		using DataHeatBalSurface::TH;
+		using DataSurfaces::Surface;
+		using DataSurfaces::TotSurfaces;
+		using DataHeatBalance::Construct;
+
+		// Return value
+		Real64 CalcSurfaceWeightedMRT;
+		
+		// Local variables
+		int SurfNum2; // surface number used in "for" loop
+		int ZoneNum2; // zone number index
+		Real64 SumAET; // Intermediate calculational variable (area*emissivity*T) sum
+		static Array1D< Real64 > SurfaceAE; // Product of area and emissivity for each surface
+		static Array1D< Real64 > ZoneAESum; // Sum of area times emissivity for all zone surfaces
+
+		// Initialize ZoneAESum for all zones and SurfaceAE for all surfaces
+		if ( FirstTimeSurfaceWeightedFlag ) {
+			FirstTimeSurfaceWeightedFlag = false;
+			SurfaceAE.allocate( TotSurfaces );
+			ZoneAESum.allocate( NumOfZones );
+			SurfaceAE = 0.0;
+			ZoneAESum = 0.0;
+			for ( SurfNum2 = 1; SurfNum2 <= TotSurfaces; ++SurfNum2 ) {
+				if ( Surface( SurfNum2 ).HeatTransSurf ) {
+					SurfaceAE( SurfNum2 ) = Surface( SurfNum2 ).Area * Construct( Surface( SurfNum2 ).Construction ).InsideAbsorpThermal;
+					ZoneNum2 = Surface( SurfNum2 ).Zone;
+					// Do NOT include the contribution of the Surface that is being surface weighted in this calculation since it will already be accounted for
+					if ( ( ZoneNum2 > 0 ) && ( SurfNum2 != SurfNum ) ) ZoneAESum( ZoneNum2 ) += SurfaceAE( SurfNum2 );
+				}
+			}
+		}
+		
+		// Calculate the sum of area*emissivity*temperature for all surfaces in the zone EXCEPT the surface being weighted
+		SumAET = 0.0;
+		for ( SurfNum2 = Zone( ZoneNum ).SurfaceFirst; SurfNum2 <= Zone( ZoneNum ).SurfaceLast; ++SurfNum2 ) {
+			if ( ( Surface( SurfNum2 ).HeatTransSurf ) && ( SurfNum2 != SurfNum ) ) {
+				SumAET += SurfaceAE( SurfNum2 ) * TH( 2, 1, SurfNum2 );
+			}
+		}
+		
+		// Now weight the MRT--half comes from the surface used for weighting (SurfNum) and the rest from the adjusted MRT that excludes this surface
+		if ( ZoneAESum( ZoneNum ) > 0.01 ) {
+			CalcSurfaceWeightedMRT = 0.5 * ( TH( 2, 1, SurfNum ) + ( SumAET / ZoneAESum( ZoneNum ) ) );
+		} else {
+			ShowWarningError( "Zone areas*inside surface emissivities are summing to zero, for Zone=\"" + Zone( ZoneNum ).Name + "\"" );
+			ShowContinueError( "As a result, MAT will be used for MRT when calculating a surface weighted MRT for this zone." );
```

Comment:
Don't want these warnings to write every timestep.  Sufficient to write this once - and add "only the first occurrence is reported" or something to that effect.

Reply:
@mjwitte Ok, I will change this as well.  It shouldn't ever happen, but I get that if it does that no one wants a 20Gb file with this message repeated over and over and over again.